lib/shopify-cli/js_deps.rb

```diff
@@ -1,28 +1,49 @@
 require 'shopify_cli'
 
-module Node
+module ShopifyCli
+  ##
+  # ShopifyCli::JsDeps ensures that all JavaScript dependencies are installed for projects.
+  #
   class JsDeps
     include SmartProperties
 
     property :ctx, accepts: ShopifyCli::Context, required: true
 
-    def self.install(ctx)
-      new(ctx: ctx).install
+    ##
+    # Proxy to ShopifyCli::JsDeps.new.install.
+    #
+    def self.install(ctx, verbose = false)
+      new(ctx: ctx).install(verbose)
     end
 
-    def yarn?
-      File.exist?(File.join(ctx.root, 'yarn.lock')) && CLI::Kit::System.system('which', 'yarn').success?
-    end
-
-    def install
+    ##
+    # Installs all of a project's JavaScript dependencies using Yarn or NPM, based on the project's settings.
+    #
+    # #### Parameters
+    # - `verbose`: whether to run the installation tools in silent mode.
+    #
+    # #### Example
+    #
+    #   ShopifyCli::JsDeps.new(ShopifyCli::Context context).install
+    #
+    def install(verbose = false)
       CLI::UI::Frame.open(ctx.message('node.js_deps.installing', yarn? ? 'yarn' : 'npm')) do
-        yarn? ? yarn : npm
+        yarn? ? yarn(verbose) : npm(verbose)
       end
       ctx.done(ctx.message('node.js_deps.installed'))
     end
 
-    def yarn
-      success = CLI::Kit::System.system('yarn', 'install', '--silent', chdir: ctx.root) do |out, err|
+    private
+
+    def yarn?
+      File.exist?(File.join(ctx.root, 'yarn.lock')) && CLI::Kit::System.system('which', 'yarn').success?
+    end
+
+    def yarn(verbose = false)
+      cmd = %w(yarn install)
+      cmd << '--silent' unless verbose
+
+      success = CLI::Kit::System.system(*cmd, chdir: ctx.root) do |out, err|
         puts out
         err.lines.each do |e|
           puts e
```

Comment:
Shouldn't these `puts` be `ctx.puts`?

Reply:
Very good point - I think they should. I'll see if there's anything keeping us from doing that and fix it otherwise. Thanks!